File: alligator/src/main/java/me/aartikov/alligator/helpers/DialogFragmentHelper.java
package me.aartikov.alligator.helpers;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.FragmentManager;

import me.aartikov.alligator.animations.DialogAnimation;

/**
 * Date: 25.03.2017
 * Time: 15:19
 *
 * @author Artur Artikov
 */

/**
 * Helper class for showing and hiding a dialog fragment.
 */
public class DialogFragmentHelper {
	private static final String TAG = "me.aartikov.alligator.DIALOG_FRAGMENT_HELPER_TAG";
	private FragmentManager mFragmentManager;

	public DialogFragmentHelper(@NonNull FragmentManager fragmentManager) {
		mFragmentManager = fragmentManager;
	}

	@Nullable
	public DialogFragment getDialogFragment() {
		DialogFragment dialogFragment = (DialogFragment) mFragmentManager.findFragmentByTag(TAG);
		if (dialogFragment == null || dialogFragment.isRemoving()) {
			return null;
		} else {
			return dialogFragment;
		}
	}

	public boolean isDialogVisible() {
		return getDialogFragment() != null;
	}

	public void showDialog(@NonNull DialogFragment dialogFragment, @NonNull DialogAnimation animation) {
		animation.applyBeforeShowing(dialogFragment);
		dialogFragment.show(mFragmentManager, TAG);
		mFragmentManager.executePendingTransactions();
		animation.applyAfterShowing(dialogFragment);
	}

	public void hideDialog() {
		DialogFragment dialogFragment = (DialogFragment) mFragmentManager.findFragmentByTag(TAG);
		if (dialogFragment == null) {
			throw new IllegalStateException("Dialog is not visible.");
		}

		dialogFragment.dismiss();
		mFragmentManager.executePendingTransactions();
	}
}

File: alligator/src/main/java/me/aartikov/alligator/helpers/DialogFragmentHelper.kt
package me.aartikov.alligator.helpers

import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentManager
import me.aartikov.alligator.animations.DialogAnimation

/**
 * Helper class for showing and hiding a dialog fragment.
 */
class DialogFragmentHelper(private val mFragmentManager: FragmentManager) {
    val dialogFragment: DialogFragment?
        get() {
            val dialogFragment = mFragmentManager.findFragmentByTag(TAG) as DialogFragment?
            return if (dialogFragment == null || dialogFragment.isRemoving) {
                null
            } else {
                dialogFragment
            }
        }
    val isDialogVisible: Boolean
        get() = dialogFragment != null

    fun showDialog(dialogFragment: DialogFragment, animation: DialogAnimation) {
        animation.applyBeforeShowing(dialogFragment)
        dialogFragment.show(mFragmentManager, TAG)
        mFragmentManager.executePendingTransactions()
        animation.applyAfterShowing(dialogFragment)
    }

    fun hideDialog() {
        val dialogFragment = mFragmentManager.findFragmentByTag(TAG) as DialogFragment?
            ?: throw IllegalStateException("Dialog is not visible.")
        dialogFragment.dismiss()
        mFragmentManager.executePendingTransactions()
    }

    companion object {
        private const val TAG = "me.aartikov.alligator.DIALOG_FRAGMENT_HELPER_TAG"
    }
}